pkg/scheduler/algorithm/priorities/resource_limits.go

```diff
@@ -43,7 +43,7 @@ func ResourceLimitsPriorityMap(pod *v1.Pod, meta interface{}, nodeInfo *schedule
 
 	// compute pod limits
 	var podLimits *schedulernodeinfo.Resource
-	if priorityMeta, ok := meta.(*priorityMetadata); ok {
+	if priorityMeta, ok := meta.(*priorityMetadata); ok && priorityMeta != nil {
```

Comment:
we could revert this file, this case could only happen in the unit test

Reply:
I think I had a discussion with someone before

Reply:
Keep this it is fine. A non-typed nil will fail the cast, but a typed nil will pass the cast and cause NPE.